cmd/influx_inspect/export/export.go

```diff
@@ -123,94 +122,90 @@ func (cmd *Command) export() error {
-		if filepath.Ext(dir) != ext {
+		if filepath.Ext(path) != "."+tsm1.TSMFileExtension {
 			return nil
 		}
 
-		relPath, _ := filepath.Rel(cmd.dataDir, dir)
+		relPath, err := filepath.Rel(cmd.dataDir, path)
+		if err != nil {
+			return err
+		}
 		dirs := strings.Split(relPath, string(byte(os.PathSeparator)))
 		if len(dirs) < 2 {
-			return fmt.Errorf("invalid directory structure for %s", dir)
+			return fmt.Errorf("invalid directory structure for %s", path)
 		}
 		if dirs[0] == cmd.database || cmd.database == "" {
 			if dirs[1] == cmd.retentionPolicy || cmd.retentionPolicy == "" {
 				key := filepath.Join(dirs[0], dirs[1])
-				files := cmd.tsmFiles[key]
-				if files == nil {
-					files = []string{}
-				}
 				cmd.manifest[key] = struct{}{}
-				cmd.tsmFiles[key] = append(files, dir)
+				cmd.tsmFiles[key] = append(cmd.tsmFiles[key], path)
 			}
 		}
 		return nil
 	})
-	if err != nil {
-		return err
-	}
-	return nil
 }
 
 func (cmd *Command) walkWALFiles() error {
-	err := filepath.Walk(cmd.walDir, func(dir string, f os.FileInfo, err error) error {
+	return filepath.Walk(cmd.walDir, func(path string, f os.FileInfo, err error) error {
 		if err != nil {
 			return err
 		}
 
 		// check to see if this is a wal file
-		prefix := tsm1.WALFilePrefix
-		ext := fmt.Sprintf(".%s", tsm1.WALFileExtension)
-		_, fileName := path.Split(dir)
-		if filepath.Ext(dir) != ext || !strings.HasPrefix(fileName, prefix) {
+		fileName := filepath.Base(path)
+		if filepath.Ext(path) != "."+tsm1.WALFileExtension || !strings.HasPrefix(fileName, tsm1.WALFilePrefix) {
 			return nil
 		}
 
-		relPath, _ := filepath.Rel(cmd.walDir, dir)
+		relPath, err := filepath.Rel(cmd.walDir, path)
+		if err != nil {
+			return err
+		}
 		dirs := strings.Split(relPath, string(byte(os.PathSeparator)))
 		if len(dirs) < 2 {
-			return fmt.Errorf("invalid directory structure for %s", dir)
+			return fmt.Errorf("invalid directory structure for %s", path)
 		}
 		if dirs[0] == cmd.database || cmd.database == "" {
 			if dirs[1] == cmd.retentionPolicy || cmd.retentionPolicy == "" {
 				key := filepath.Join(dirs[0], dirs[1])
-				files := cmd.walFiles[key]
-				if files == nil {
-					files = []string{}
-				}
 				cmd.manifest[key] = struct{}{}
-				cmd.walFiles[key] = append(files, dir)
+				cmd.walFiles[key] = append(cmd.walFiles[key], path)
 			}
 		}
 		return nil
 	})
-	if err != nil {
-		return err
-	}
-	return nil
 }
 
-func (cmd *Command) writeFiles() error {
+func (cmd *Command) write() error {
 	// open our output file and create an output buffer
-	var w io.WriteCloser
-	w, err := os.Create(cmd.out)
+	f, err := os.Create(cmd.out)
 	if err != nil {
 		return err
 	}
-	defer w.Close()
+	defer f.Close()
+
+	// Because calling (*os.File).Write is relatively expensive,
+	// and we don't *need* to sync to disk on every written line of export,
+	// use a sized buffered writer so that we only sync the file every megabyte.
+	bw := bufio.NewWriterSize(f, 1024*1024)
```

Comment:
Can this not just be `w := bufio....` and then we don't need the `var w = bw` below?